vocabulary-draft/continuous-integration-pipeline-events.md

```diff
@@ -0,0 +1,26 @@
+# Continuous Integration Events
+
+These events are related to **Continuous Integration(CI)** activities. CI usually include activities such as building, testing, packaging and releasing software artifacts. 
+
+The following events represent concrete Tasks that are associated with the execution of CI pipelines:
+
+- **Build Queued**: a Build task has been queued, this build process usually is in charge of producing a binary from source code
+- **Build Started**: a Build task has started 
+- **Build Finished**: a Build task has finished, the event will contain the finished status, success, error or failure
+
+The following Test events are defined in two separate categories **Test Case** and **Test Suite**. A **Test Case** is the smallest unit of testing that the user wants to track. A **Test Suite** is a collection of test case executions and/or other test suite executions. **Test Cases** executed, and Test Suites are for grouping purposes. For this reason, **Test Cases** can be queued. 
+
+- **Test Case Queued**: a Test task has been queued, and it is waiting to be started
+- **Test Case Started**: a Test task has started
+- **Test Case Finished**: a Test task has finished, the event will contain the finished status, success, error or failure
+- **Test Suite Started**: a Test task has started
+- **Test Suite Finished**: a Test task has finished, the event will contain the finished status, success, error or failure
+
+Finally, events needs to be generated for the output of the pipeline such as the artifacts that were packaged and released for others to use. 
+
+- **Artifact Packaged**: an artifact has been packaged for distribution, this artifact is now versioned with a fixed version
+- **Artifact Released**: an artifact has been released and it can be advertised for others to use
+
+CI Events MUST include the following attributes:
+ - **Event Type**: the type is restricted to include `CD.CI.**` prefix. For example `CD.CI.Build.Queued` or `CD.CI.Artifact.Packaged`
+ - **Artifact Id**: the unique identifier of the artifact that the event is refering to. 
```

Comment:
What would this be for build and test events?

Reply:
@erkist you mean the `Artifact Id`? I was thinking the artifact that was built and the artifact that was tested. When working with Services or Libraries, we tend to build the service or the library binary and same with testing, we consider all the test suites related to the artifact that we are building, does this make sense? 
Side Note: Should we use artifact or artefact? I always have issues with those.

Reply:
@salaboy That makes sense, but it causes a problem for the events that state that a build has been queued or started? No artifact has been produced yet, so there is no ID to provide for the mandatory attribute.

Reply:
@erkist ok I think that this might be also different for different tools. Some tools allow you to define what is the expected output of the pipeline or you can define that just as metadata of the pipeline itself. So other tools can pick up these artifacts and do things. But I totally understand that you might have pipelines also that just build stuff without knowing. I wonder if PipelineRun output or Pipeline Output is something that we need. @afrittoli does tekton has some of these concepts?

Reply:
Agree, I @salaboy wants to have something to tie the events together but that will be difficult if e.g. the artifact id would be a hash of the built artifact.

Reply:
I'm not sure I follow. My point is: In the events "Build Queued" and "Build Started", it is hard to provide the ID of the artifact that has not yet been built. This is especially true if the build then fails, and no artifact is built.

Reply:
To be clear, I agree with you @erkist.

I think @salaboy assumes that we always know the id of the artifact to build before we trigger the build job. I could think of many scenarios where the artifact ID is calculated during the build process.

Reply:
I'd say that CI events SHOULD have a means to refer to what _source change_ that triggered it, maybe that is what you really mean @salaboy ? I agree with @erkist and @m-linner-ericsson that for all CI events preceeding the actual build we probably don't know what the artifact id will be. So the build queued and build started events mentioned here cannot have artifact id as a mandatory field

Reply:
@erkist @m-linner-ericsson @e-backmark-ericsson we are all in the same page now, let's not add the Artifact ID as a mandatory field for these events. I will change that now.